shaders/src/main/glsl/samples/310es/stable_quicksort.frag

```diff
@@ -0,0 +1,114 @@
+#version 310 es
+
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+precision highp float;
+
+layout(location = 0) out vec4 _GLF_color;
+
+uniform vec2 injectionSwitch;
+
+uniform vec2 resolution;
+
+struct QuicksortObject{
+    int numbers[10];
+};
+
+QuicksortObject obj;
+
+ void swap(int i, int j) {
+     int temp = obj.numbers[i];
+     obj.numbers[i] = obj.numbers[j];
+     obj.numbers[j] = temp;
+ }
+
+ // Since "partition" is the preserved word, we add prefix to this function name to prevent an error.
+ int performPartition(int l, int h) {
+     // The rightmost element is chosen as a pivot.
+     int pivot = obj.numbers[h];
+     int i = (l - 1);
+  
+     for (int j = l; j <= h - 1; j++) {
+         if (obj.numbers[j] <= pivot) {
+             i++;
+             swap(i, j);
+         }
+     }
+     swap(i + 1, h);
+     return (i + 1);
+ }
+
+ void quicksort() {
+     int l = 0, h = 9;
+     int stack[10];
+     int top = -1;
+  
+     stack[++top] = l;
+     stack[++top] = h;
+  
+     while (top >= 0) {
+         h = stack[top--];
+         l = stack[top--];
+
+         int p = performPartition(l, h);
+         if (p - 1 > l) {
+             stack[++top] = l;
+             stack[++top] = p - 1;
+         }
+         if (p + 1 < h) {
+             stack[++top] = p + 1;
+             stack[++top] = h;
+         }
+     }
+ }
+
+void main() {
+    // Initialize decreasing values to an array starting from 10.
+    for (int i = int(injectionSwitch.x); i < 10; i ++) {
```

Comment:
Should we take this opportunity to remove the use of injectionSwitch, since all uses could be done by GraphicsFuzz anyway, and these forced uses might prevent bugs from being found?

Reply:
Done.